classes/Kohana/Log.php

```diff
@@ -23,9 +23,24 @@ class Kohana_Log {
 	const DEBUG     = LOG_DEBUG;    // 7
 
 	/**
+	 * Numeric log level to string lookup table.
+	 * @var array
+	 */
+	protected $_log_levels = array(
+		LOG_EMERG   => \Psr\Log\LogLevel::EMERGENCY,
+		LOG_ALERT   => \Psr\Log\LogLevel::ALERT,
+		LOG_CRIT    => \Psr\Log\LogLevel::CRITICAL,
+		LOG_ERR     => \Psr\Log\LogLevel::ERROR,
+		LOG_WARNING => \Psr\Log\LogLevel::WARNING,
+		LOG_NOTICE  => \Psr\Log\LogLevel::NOTICE,
+		LOG_INFO    => \Psr\Log\LogLevel::INFO,
+		LOG_DEBUG   => \Psr\Log\LogLevel::DEBUG,
+	);
+
+	/**
 	 * @var  boolean  immediately write when logs are added
 	 */
-	public static $write_on_add = FALSE;
+	protected $write_immediately = FALSE;
```

Comment:
Not sure why would they need, but not sure either why restrict? What do we loose?